autofs-5.1.5 - allow period following macro in selector value

From: Ian Kent <raven@themaw.net>

If a ${} macro is used in a selector (such as hostd) the parser doesn't
permit a period to follow it (such as when the value is a host name).

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG         |    1 +
 modules/amd_tok.l |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- autofs-5.0.7.orig/CHANGELOG
+++ autofs-5.0.7/CHANGELOG
@@ -326,6 +326,7 @@
 - make expire remaining log level debug.
 - workaround getaddrinfo(3) ai_canonname bug
 - improve hostname lookup error logging.
+- allow period following macro in selector value.
 
 25/07/2012 autofs-5.0.7
 =======================
--- autofs-5.0.7.orig/modules/amd_tok.l
+++ autofs-5.0.7/modules/amd_tok.l
@@ -95,7 +95,7 @@ V6MASK		(12[0-8]|1[0-1][0-9]|[1-9][0-9]|
 
 FOPT		(({QSTR}|{FSTR}|{MACRO})+)
 OPTS		({OSTR}(=({VSTR}|{MACRO})+)?)
-SOPT		(({SSTR}|{QSTR}|{MACRO})+)
+SOPT		({SSTR}|{QSTR}|{MACRO}(\.|{SSTR}|{QSTR}|{MACRO})+)
 NOPT		({SSTR}|(({IP4ADDR}(\/{V4MASK})?)|({IP6ADDR}(\/{V6MASK})?)))
 
 MAPOPT		(fs|type|maptype|pref|sublink|cache)
